Assignment2/Node.java

```diff
@@ -0,0 +1,47 @@
+package Assignment2;
+
+public class Node<T> {
+	private T key;
+	Node<T> left;
+	Node<T> right;
+	
+	public Node() {
+		
+	}
+	
+	public Node(T key) {
+		this.key = key;
+	}
+
+	public Node(T key, Node<T> left, Node<T> right){
+		this.key = key;
+		this.left = left;
+		this.right = right;
+	}
+	
+	public T getKey(){
+		return this.key;
+	}
+
+	public void setKey(T newKey) {
+		this.key = newKey;
+	}
+
+	@Override
+	public boolean equals(Object obj) {

```

Comment:
You should never add an equals method without adding a hashcode method.
https://stackoverflow.com/questions/27581/what-issues-should-be-considered-when-overriding-equals-and-hashcode-in-java